test/DumberComponent_test.js

```diff
@@ -0,0 +1,33 @@
+const React = require('react');
+const { shallow } = require('enzyme');
+const sinon = require('sinon');
+
+const Noop = (props) => { return <p>Noop</p> };
+const DumberComponent = require('../components/DumberComponent');
+
+describe('<DumberComponent />', () => {
+  let wrapper, spy;
+
+  beforeEach(() => {
+    spy = sinon.spy();
+    wrapper = !DumberComponent.prototype ?
+      shallow(<Noop />) : shallow(<DumberComponent handleClick={spy} />);
+  });
+
+  it('should not be a class-based component', () => {
+    const tryToGetState = () => { wrapper.state(); }
+    expect(tryToGetState).toThrow(
+      'ShallowWrapper::state() can only be called on class components'
```

Comment:
This error output actually comes from enzyme, and the toThrow method is checking for that particular message as a way to ensure that they are getting **that** error and not another. But let me see if I can add a custom message. Another possibility would be to throw a custom error in the `tryToGetState` fn above.